tfjs-backend-webgpu/src/backend_webgpu.ts

```diff
@@ -102,7 +101,7 @@ export class WebGPUBackend extends KernelBackend {
   commandQueue: GPUCommandEncoder[];
 
   private commandQueueOwnedIds = new WeakSet<DataId>();
-  private binaryCache: {[key: string]: WebGPUBinary};
+  private programCache: Map<number, webgpu_program.WebGPUProgram> = new Map();
```

Comment:
Another (very possibly bad) idea to avoid the need for a hash is to make a big nested table of Maps such that the lookup would do `programCache.get(x).get(y).get(z)`. No idea how well this would perform. The depth could be reduced a little bit by combining some numbers into a single key (e.g. `width * 0x4000000 + height` if you know width is &lt; 2<sup>26</sup>).